src/main/scala/com/high-performance-spark-examples/ml/CustomPipeline.scala

```diff
@@ -122,7 +122,10 @@ class SimpleIndexer(override val uid: String)
     import dataset.sparkSession.implicits._
     val words = dataset.select(dataset($(inputCol)).as[String]).distinct
       .collect()
-    new SimpleIndexerModel(uid, words)
+    val model = new SimpleIndexerModel(uid, words)
+    model.set(inputCol, $(inputCol))
```

Comment:
Thanks for catching this! The proper fix is likely to use the `copyValues` function from the Params trait.